### ISSUE TITLE
Binding `Content` Property of `ContentView` to a nested property woud result in reparenting error on Android

### ISSUE THREAD

**Issue by rbakhshi** [dotnet/maui#12261]
### Description

Checkout the reproduction repository for sample code. But basically if you bind the `content` to something like `Object.Property` and have `Object` changed to something else, upon re-rending the `ContentView` you will get 

```
Java.Lang.IllegalStateException
  Message=The specified child already has a parent. You must call removeView() on the child's parent first.
```

### Steps to Reproduce

1. Checkout the code from repo (branch: `reparenting-issue`)
2. Run the app in Android emulator
3. Click on `Clicked x times` button 3 times
4. The app crashes immediately upon 3rd click

### Link to public reproduction project repository

https://github.com/rbakhshi/bugreporting/tree/reparenting-issue

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android Api 32

### Did you find any workaround?

In `MainPage.Xaml` if you comment the block from line 12-14 and instead uncomment the block below it (from line 16-21) it will work.

Basically avoid navigational property access in binding of content and instead bind to each property one at a time:

This fails:
```xml
 <ControlTemplate x:Key="DynamicContent">
     <ContentView Content="{Binding BindingContext.Content.Header, Source={RelativeSource AncestorType={x:Type ContentPage}}}"/>
</ControlTemplate>
```

This works:
```xml
<ControlTemplate x:Key="DynamicContent">
    <ContentView BindingContext="{Binding BindingContext, Source={RelativeSource AncestorType={x:Type ContentPage}}}">
          <ContentView Content="{Binding ContentHeader}"/>
    </ContentView>
</ControlTemplate>
```

**Comment by BagavathiPerumal** [comment]
This issue is not reproduced in the latest Maui version(10.0.11). I have attached a video for reference.

https://github.com/user-attachments/assets/222268e7-29e8-4527-a5cb-e99c39e181d4
